consumer/pdatautil/metricsbuilder.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdatautil
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/internal/data"
+)
+
+// MetricsBuilder facilitates building pdata.Metrics in receivers.  It is meant
+// to be much easier and more fluent than than using pdata.Metrics directly.
+// All of the methods on it except for Flush() will return the same instance of
+// MetricsBuilder as a return value, allowing you to chain method calls easily,
+// similar to the Java builder pattern.
+//
+// The MetricsBuilder is designed for cases where receivers are generating
+// metrics from scratch, where generally you will have a single datapoint per
+// metric/label combination.
```

Comment:
You seem to mean one data point per metric based on the code, which I think is a severely limiting factor. It's pretty common to have multiple data points per metric isn't it?

Reply:
This is an area that I need to optimize in the builder -- by making it reuse `pdata.Metric` instances where the metric name and type are the same.  Even as it is now, you can add any number of datapoints per metric you want, but they will just be separate `pdata.Metric` instances which is kind of wasteful.

The one thing that is going to be very rare when generating metrics "from scratch" is having datapoints with the same metric name/label-set but different timestamps.  Technically you can do this with the builder as is by changing the Timestamp field and making new metrics, if somebody really needs that.

Reply:
I think generating metrics with the same label keys and different label values for each point would be relatively common. 
 This is done this w/ per-cpu in the proposed docker stats receiver.  The model was also created for this use case afaict.

Reply:
I think you are thinking of the OTLP format where label keys and values are split up -- the `pdata.Metrics` format is more like our SignalFx datapoints where it is just a key/value pair all in one.  In `pdata.Metrics` you can group all metrics with the same name and type into a single `Metric` object, which is what I need to do.

Reply:
Is pdata not a wrapper for otlp and without format https://github.com/open-telemetry/opentelemetry-collector/blob/master/consumer/pdata/generated_metrics.go#L530 ? I see this limitation is in its datapoint api though.